File: src/main/java/com/example/librarymanagement/dto/mapper/ErrorMapper.java
package com.example.librarymanagement.dto.mapper;

import com.example.librarymanagement.dto.ErrorDto;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class ErrorMapper {
    public ErrorDto toErrorDto(HttpStatus status, List<String> message) {
        ErrorDto errorDto = new ErrorDto();
        errorDto.setStatus(status.value());
        errorDto.setMessage(message);
        return errorDto;
    }

    public ErrorDto toErrorDto(HttpStatus status, String message) {
        ErrorDto errorDto = new ErrorDto();
        errorDto.setStatus(status.value());
        errorDto.setMessage(List.of(message));
        return errorDto;
    }
}

File: src/main/java/com/example/librarymanagement/dto/mapper/ErrorMapper.kt
package com.example.librarymanagement.dto.mapper

import com.example.librarymanagement.dto.ErrorDto
import org.springframework.http.HttpStatus
import org.springframework.stereotype.Component

@Component
class ErrorMapper {
    fun toErrorDto(status: HttpStatus, messages: List<String>): ErrorDto {
        return ErrorDto(
            status = status.value(),
            messages = messages
        )
    }

    fun toErrorDto(status: HttpStatus, message: String): ErrorDto {
        return ErrorDto(
            status = status.value(),
            messages = listOf(message)
        )
    }
}